### PR TITLE
[SW-2062] Verify that training frame has all columns constants using Spark instead of H2O error

### PR DESCRIPTION
**Pull request by jakubhava**
@mn-mikke promised this change a while ago. As you suggested, we can check the validity even before we start the training

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala [hunk 1]
```diff
@@ -52,6 +53,11 @@ trait H2OAlgoCommonUtils extends EstimatorCommonUtils {
     }
 
     val featureColumns = getFeaturesCols().map(sanitize).map(col)
+
+    if (dataset.select(featureColumns: _*).distinct().count() == 1) {
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:57]: Isn't [this H2O option](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/ignore_const_cols.html#ignore-const-cols) important for make the condition?

Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:57]: Good find! But I don't think so; we just need to check whether all columns are constants. If this option is applied, H2O will have no columns to use as features and we throw the same exception which I thing is correct. WDYT?

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala [hunk 2]
```diff
@@ -52,6 +53,11 @@ trait H2OAlgoCommonUtils extends EstimatorCommonUtils {
...
+      throw new IllegalArgumentException(s"H2O could not use any of the specified feature" +
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:58]: nit `feature` ->  `features`

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala [hunk 3]
```diff
@@ -52,6 +53,11 @@ trait H2OAlgoCommonUtils extends EstimatorCommonUtils {
...
+        s" columns: '${getFeaturesCols().mkString(", ")}' because they are all constants. H2O ignores constant columns.")
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:59]: `H2O ignores constant columns.` -> `H2O requires at least one non-constant column.`

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:57]: It seems that you right. The comment inside H2O code says:
`Check that at least some columns are not-constant and not-all-NAs`

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala (hunk 2 above)
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala:58]: 👍

**Pull request merged**